frontend/src/metabase/common/components/AggregationPicker/AggregationPicker.tsx

```diff
@@ -220,7 +220,12 @@ export function AggregationPicker({
         withName
         startRule="aggregation"
         header={<ExpressionWidgetHeader onBack={closeExpressionEditor} />}
-        onChangeClause={handleClauseChange}
+        onChangeClause={(name, clause) => {
+          handleClauseChange(
+            name,
+            clause as Lib.AggregationClause | Lib.ExpressionClause,
```

Comment:
It's a pity that `AggregationClause` does not _extend_ `ExpressionClause` - then we would not need a cast

Reply:
Done: 105e9a2e274f3f529e44c5794ee120a662f286e + 73cccadfe5be077c94a1cacdd7ffd73d7a10ce80 + e470aff03565da3acc43e87b8a80a84b2d78b20 + 2676fcb6c3f04452c6a8688e9edbea80f6ef0de

But we still need casts because:
- `FilterClause` should never be used with `withExpressionName`
- `AggregationClause` will not work with `Lib.filter()`